github-action/src/main.ts

```diff
@@ -90,47 +90,51 @@ export async function runMain(): Promise<void> {
 			return;
 		}
 
-		const upResult = await core.group('🏃 start container', async () => {
-			const args: DevContainerCliUpArgs = {
-				workspaceFolder,
-				additionalCacheFroms: cacheFrom,
-				skipContainerUserIdUpdate,
-				userDataFolder,
-			};
-			const result = await devcontainer.up(args, log);
-			if (result.outcome !== 'success') {
-				core.error(
-					`Dev container up failed: ${result.message} (exit code: ${result.code})\n${result.description}`,
-				);
-				core.setFailed(result.message);
-			}
-			return result;
-		});
-		if (upResult.outcome !== 'success') {
-			return;
-		}
-
-		const execResult = await core.group(
-			'🚀 Run command in container',
-			async () => {
-				const args: DevContainerCliExecArgs = {
+		if (runCommand) {
+			const upResult = await core.group('🏃 start container', async () => {
+				const args: DevContainerCliUpArgs = {
 					workspaceFolder,
-					command: ['bash', '-c', runCommand],
-					env: inputEnvsWithDefaults,
+					additionalCacheFroms: cacheFrom,
+					skipContainerUserIdUpdate,
 					userDataFolder,
 				};
-				const result = await devcontainer.exec(args, log);
+				const result = await devcontainer.up(args, log);
```

Comment:
Are these `exec` and `up` commands wrappers in this repo?  If so, is there a chance we could move those back to the CLI itself? It would be useful internally to be able to run `up` with a set of arguments (for example, in the `features test <...>` command.

Reply:
Yes - see https://github.com/devcontainers/ci/blob/main/common/src/dev-container-cli.ts

Re moving to the CLI - I originally thought that the CLI package could be set up so that it could be used either as a CLI or as a package that can be used from JS/TS. IIRC, @chrmarti wanted to keep it as a pure CLI so that everyone got a consistent experience, but you may want to check with him when he's back